daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -711,7 +714,7 @@ object SBuiltin {
           choiceId = choiceId,
           optLocation = machine.lastLocation,
           consuming = consuming,
-          actingParties = actors,
+          actingParties = optActors.getOrElse(ctrls),
```

Comment:
@remyhaemmerle-da @bitonic Can you please confirm whether this is the right place to fill in the controllers for missing actors.

Reply:
looks good.